tfe/provider.go

```diff
@@ -282,36 +282,71 @@ func getClient(tfeHost, token string, insecure bool) (*tfe.Client, error) {
 // cliConfig tries to find and parse the configuration of the Terraform CLI.
 // This is an optional step, so any errors are ignored.
 func cliConfig() *Config {
-	config := &Config{}
-
-	// Detect the CLI config file path.
-	configFilePath := os.Getenv("TERRAFORM_CONFIG")
+	mainConfig := &Config{}
+	combinedConfig := &Config{}
+
+	// There might be credentials in the main CLI config file (manually entered)
+	// AND/OR the credentials file (auto-configured by terraform login or a
+	// credentials helper), so we need to consult both. As per the behavior of
```

Comment:
This doesn't work with a credentials helper, no? 

FWIW we could support them, the logic is here: https://github.com/hashicorp/terraform-svchost/blob/f050f53b97348772833a8f7fc1b297ec71f9fc7c/auth/helper_program.go

But this PR is already 💯 without all that, for sure 👍🏻 I'd just amend this comment.

Reply:
Oh! I thought credentials helpers used the credentials file as a cache, though now I've forgotten where I would have gotten that idea. 

I'll just amend the comment to only refer to login. 👍🏼